testsuite/forge/src/backend/k8s/cluster_helper.rs

```diff
@@ -615,32 +615,47 @@ fn dump_helm_values_to_file(helm_release_name: &str, tmp_dir: &TempDir) -> Resul
 
 pub async fn create_management_configmap(kube_namespace: String, keep: bool) -> Result<()> {
     let kube_client = create_k8s_client().await;
-    let pp = PostParams::default();
 
-    let namespaces_api: Api<Namespace> = Api::all(kube_client.clone());
-    let namespace = Namespace {
-        metadata: ObjectMeta {
-            name: Some(kube_namespace.clone()),
-            ..ObjectMeta::default()
-        },
-        spec: None,
-        status: None,
-    };
-
-    if let Err(KubeError::Api(api_err)) = namespaces_api.create(&pp, &namespace).await {
-        if api_err.code == 409 {
-            info!(
-                "Namespace {} already exists, continuing with it",
-                &kube_namespace
-            );
-        } else {
-            bail!(
-                "Failed to use existing namespace {}: {:?}",
-                &kube_namespace,
-                api_err
-            );
-        }
-    };
+    // try to create a new namespace
+    // * if it errors with 409, the namespace exists already and we should use it
+    // * if it errors with 403, the namespace is likely in the process of being terminated, so try again
```

Comment:
Where are we checking for this error code? Is the documentation up to date?

Reply:
We're explicitly checking for 409 since we can continue despite it, but the rest we're catching and retrying. https://docs.rs/kube/latest/kube/core/struct.ErrorResponse.html